lib/modules/manager/gomod/extract.ts

```diff
@@ -62,8 +67,22 @@ export function extractPackageFile(
       const line = lines[lineNumber];
       const goVer = line.startsWith('go ') ? line.replace('go ', '') : null;
       if (goVer && semver.validRange(goVer)) {
-        const dep = getGoDep(lineNumber, goVer);
+        const dep = getGoDep(lineNumber, goVer, 'go-mod-directive');
+        deps.push(dep);
+        continue;
+      }
+      const goToolVer = line.startsWith('toolchain go')
+        ? line.replace('toolchain go', '')
+        : null;
+      if (goToolVer && semver.valid(goToolVer)) {
+        const dep = getGoDep(
+          lineNumber,
+          goToolVer,
+          undefined,
+          'golangtoolchain',
```

Comment:
@rarkins @secustor should we use `toolchain` only? 🤔

Reply:
I agree, it should be `toolchain` as the gomod context should be clear from the manager.